pkg/sqlcache/db/client.go

```diff
@@ -102,15 +103,49 @@ func (c *client) withTransaction(ctx context.Context, forWriting bool, f WithTra
 	return nil
```

Comment:
I realize it's not part of this PR, but the lines 
```
err = c.commit(ctx, tx)
	if err != nil {
		// When the context.Context given to BeginTx is canceled, then the
		// Tx is rolled back already, so rolling back again could have failed.
		return err
	}
	return nil
```
can be folded into `return c.commit(ctx, tx)` with the comment preceding the call.